### ISSUE TITLE
Cyme2DSS.py conversion issue, OpenDSS erroring on single phase lines

### ISSUE THREAD

**Issue by jaimiosyncrasy** [GRIDAPPSD/CIMHub#19]
I'm using the CYME2DSS.py script in the converters folder of this repo to convert a CYMEDIST .sxst file to OpenDSS files. The conversion completed, but when solving the circuit files in OpenDSS I receive the following errors:
![conversion_error_wiredata](https://user-images.githubusercontent.com/6557952/134245754-fd359b68-6f85-4e1a-9324-371b2256bb00.jpg)

My suspicion: I noticed OpenDSS errors out on whatever is the first single phase conductor in the network.dss file, so it is likely that the conversion of single phase lines was done incorrectly. Perhaps OpenDSS thinks the single line is 3-phase so is pulling empty fields from the non-existent phases and erroring when it doesn't understand what wire " " is

Here is a subset of the xml that corresponds to the first single phase line in the network.dss file. Certain field values are redacted due to an NDA, but the code structure is intact
[line_redacted_sxst_sample.txt](https://github.com/GRIDAPPSD/CIMHub/files/7206160/line_redacted_sxst_sample.txt)

Once parsed this line is translated into the following entry in network.dss, which is creating the errors in the screenshot
`new line.PRIOH-[LineKey] bus1=[NodeID1] bus2=[NodeID2] phases=1 spacing=OH_HORIZONTAL_1P_88 wires=[OH_4_CU_C OH_4_CU_C ] length=[Value] units=ft`

**Comment by temcdrm** [comment]
This line appears to reference a single-phase construction (spacing) with two wires, one for the phase and the other for the neutral, which looks okay.  However, the message suggests OpenDSS can't find the data for OH_4_CU_C, which should contain details referenced in your SXST file extract, e.g., <PhaseConductorIDA>4_CU_C</PhaseConductorIDA> and <NeutralConductorID1>4_CU_C</NeutralConductorID1>. Can you search the exported OpenDSS files for anything that looks like "new wiredata.OH_4_CU_C"? 

Another thing to verify is that [NodeID1] and [NodeID2] both end in .1, because the single-phase line appears to be on phase A.

**Comment by jaimiosyncrasy** [comment]
Thank you for the quick reply. However I believe the data for OH_4_CU_C is accessible. Let's call X the line of code in OwnerID_network.dss that OpenDSS errored out on as it parses top to bottom. It did not error out on any lines <X, and those lines all say phases=3 and some of those earlier lines include OH_4_CU_C as one the types of three phases. Further, the catalog.dss file does include wire info for 4_CU_C, labeled as 'new wiredata.OH_4_CU_C', so I believe the converter does pickup the conductor specs from the .sxst

When I delete off he line in network.dss that is erroring out, OpenDSS errors out at the next later line that has less than three phases, i.e. sometimes errors at phases=1, sometimes phases=2. Most of the time the conductor that errors out is type is 4_CU_C, but one time it is 6_CU_C instead. 6_CU_C has a ''new wiredata.OH_6_CU_C'' entry in catalog.dss too. Thus I am more suspicious about OpenDSS handling <3 phase lines rather than conductor specs missing.

Confirming that [NodeID1] and [NodeID2] have the .1 (or in 2 phase case have .1.2). Also, I'm assuming it's ok for NodeID1 and NodeID2 to have underscores and dashes? For example bus1=123_45-678.1

**Comment by temcdrm** [comment]
How many phases, and x/y positions, are defined on the spacing OH_HORIZONTAL_1P_88?

**Comment by jaimiosyncrasy** [comment]
Found the issue! You're suspicion about the spacing was spot on. In the catalog.dss, I see `new linespacing.OH_HORIZONTAL_1P_88 nconds=4 nphases=3 x=[-1.77408 0 1.77408 0] h=[12.192000 12.192000 12.192000 12.192000] units=m', So while the line in network.dss that uses this line spacing has 'phases=1', in catalog.dss we have the spacing associated with 3 phases, which is a mismatch. I traced the issue to these lines of code in the cyme2dss.py parser that expects the spacing equipment ID to specify the number of phases with '-1PH-' format, but my .sxst file has equipment ID format '_1P_'. Instead of erroring out when it cant find this format, the 'else' statement just assigns 3 phases (recommend changing this to elif '-3PH-' with an else that errors out with a useful message). 
Attached is the snippet of the parser where this happens. I have just modified that portion of the code to match the format of my .sxst and it now works

![parsing_phases](https://user-images.githubusercontent.com/6557952/134759046-fbb50620-e829-4e25-a87f-86529dc05fa6.jpg)

**Comment by jaimiosyncrasy** [comment]
I now have another issue that arises when OpenDSS tries to solve the power flow. See attached please. Would appreciate any recommendations on what may be this issue
![error_about_source](https://user-images.githubusercontent.com/6557952/134759137-c31d8d44-bb7d-49b1-a373-978aef762e86.jpg)

**Comment by temcdrm** [comment]
The message about threads does not seem familiar.  Presumably you are using version 8, which I don't have much experience with.  

In _Oakland.edits_, there could be invalid characters in the bus names; please consult the OpenDSS manual to make sure that's not the case. The reference to a _ThreePhaseVoltageSource_ component does not seem familiar, at least from version 7 of OpenDSS.

**Comment by jaimiosyncrasy** [comment]
Hi Tom, that you did not recognize the message made me realize that indeed that error comes from another software that isnt OpenDSS. Mistakenly I had pasted an error message from another software into Oakland.edits. Apologies about that. 

From there I had one more parsing error in which one of the line spacing Equipment IDs (OH_HORIZONTAL_1P_88) errored out because the average phase conductor height and average neutral conductor height were given in the .sxst as equal heights, which resulted in OpenDSS saying "conductor 1 and 2 occupy the same space". So I corrected the heights in the .sxst file.

OpenDSS now successfully solves power flow and plots the circuit. The power flow result looks erroneous but that is a distinct issue from the cym2dss.py parser that I'll come back to later. In the meantime I'll close this github issue.

Thank you so much for your help Tom!
